test/functional/api/vim_spec.lua

```diff
@@ -1036,5 +1036,13 @@ describe('api', function()
       eq(expected, nvim("list_uis"))
     end)
   end)
+  describe('nvim_get_keyboard_mods_state', function()
+    it('give empty dictionary for testing as they are off in CI', function()
+      -- For this test, if you are testing on your PC, turn the capslock,
+      -- Numlock and Scrollock off, this tests the presense of these mods
+      -- and on the environment of testing they are supposed to be turn off
+      eq({CapsLock = false, NumLock = false, ScrollLock = false}, meths.get_keyboard_mods_state())
+    end)
```

Comment:
I think to not have failures which are doing nothing more then annoy user it is better to check that

1. Result of the method is either one of the known errors or a dictionary.
2. If result is a dictionary then it contains only keys from a known set of three keys.
3. If result is a dictionary then all values attached to keys are booleans.

And no specific values.

Also unless CI has X11 libraries available and X server running (so connecting to it works), should not that produce error?